.github/workflows/pytest.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will install Python dependencies and run tests with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
```

Comment:
You're going to need to install minizinc here

Reply:
Is it just pip install minizinc or I need to download from the minizinc github?

Reply:
The library is fine, you need to install the minizinc toolchain on the VM itself. MiniZinc is distributed as a [snap](https://snapcraft.io/minizinc), so maybe start there? You could also try making a [docker container action](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action), since then you can extend from the [minizinc docker image](https://hub.docker.com/r/minizinc/minizinc/tags) (like our [current docker setup](https://github.com/TechlauncherFireApp/backend/blob/main/Dockerfile)). I don't have any experience with that though, so my help would be limited beyond writing the `Dockerfile` itself.

Reply:
I attempted to install Snap and then use it to install MiniZinc in the pipeline. However, I'm uncertain about how to verify if it's running correctly. Additionally, some Python tests that fail in the pipeline run successfully on my local development environment, as well as for other team members. When I tried running the pipeline locally, it also passed. I'm unsure about how to proceed with debugging this issue. How do you typically approach debugging errors in the pipeline? My current idea is to put some try catches and push it to have more info about the errors. @BenMMcLean

Reply:
Hmm, I'm not sure actually. It also works locally for me. I say go ahead with the try catches, but experiment in a separate branch with a different pull request (just so you don't mess up this one). I'll also try.